qcengine/programs/gcp.py

```diff
@@ -0,0 +1,270 @@
+"""Compute geometrical counterpoise correction using Kruse & Grimme's GCP executable."""
+
+import os
+import pathlib
+import pprint
+import re
+import socket
+import sys
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple
+
+import numpy as np
+import qcelemental as qcel
+from qcelemental.models import AtomicResult, FailedOperation, Provenance
+from qcelemental.util import safe_version, which
+
+from ..exceptions import InputError, UnknownError
+from ..util import execute
+from .model import ProgramHarness
+
+if TYPE_CHECKING:
+    from qcelemental.models import AtomicInput
+
+    from ..config import TaskConfig
+
+
+pp = pprint.PrettyPrinter(width=120, compact=True, indent=1)
+
+
+class GCPHarness(ProgramHarness):
+
+    _defaults = {
+        "name": "GCP",
+        "scratch": True,
+        "thread_safe": True,
+        "thread_parallel": False,
+        "node_parallel": False,
+        "managed_memory": False,
+    }
+    version_cache: Dict[str, str] = {}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    @staticmethod
+    def found(raise_error: bool = False) -> bool:
+        return which(
+            "gcp",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install via `conda install gcp -c psi4`.",
+        )
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        version = None
+        which_prog = which("gcp")
+        if which_prog not in self.version_cache:
+            # option not (yet) available, instead find in help output
+            command = [which_prog, "-version"]
+            import subprocess
+
+            proc = subprocess.run(command, stdout=subprocess.PIPE)
+            for ln in proc.stdout.decode("utf-8").splitlines():
+                if re.match(".*Version", ln):
+                    print(ln.split())
+                    version = ln.split()[2]
+
+            if version is None:
+                raise UnknownError(f"Could not identify GCP version")
+            # pending upcoming gcp update
+            # self.version_cache[which_prog] = safe_version(
+            #     proc.stdout.decode("utf-8").strip())
+            self.version_cache[which_prog] = version
```

Comment:
```suggestion
            self.version_cache[which_prog] = safe_version(version)
```

yes, go ahead with safe_version, please